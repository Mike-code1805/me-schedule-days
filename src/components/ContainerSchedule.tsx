import React from 'react';
import styles from '../styles/containerSchedule/containerSchedule.modules.css';

export interface Props {
  title?: string;
  backgroundColor?: string;
  idValue?: string;
  onClickId?: any;
}

export const ContainerSchedule = ({
  title = 'ProgramaciÃ³n I',
  backgroundColor = 'red',
  idValue,
  onClickId,
}: Props) => {
  return (
    <div
      onClick={() => onClickId(idValue)}
      className={styles["containerSchedule"]}
      style={{ background: backgroundColor }}
    >
      <h1 className={styles["containerSchedule_text"]}>{title}</h1>
    </div>
  );
};
