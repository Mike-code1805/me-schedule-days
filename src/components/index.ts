import { Schedule as ScheduleComponent } from './Schedule';
import { ContainerSchedule } from './ContainerSchedule';
import { ScheduleHOCProps } from '../interfaces/schedules.interfaces';
import { DayContainerSchedule } from './DayContainerSchedule';

export const Schedule: ScheduleHOCProps = Object.assign(ScheduleComponent, {
  Container: ContainerSchedule,
  DayContainer: DayContainerSchedule,
});
