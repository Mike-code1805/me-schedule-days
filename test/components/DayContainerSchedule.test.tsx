import React from 'react';
import renderer from 'react-test-renderer';
import { Schedule } from '../../src/components/index';
import { data } from '../data/courses';

describe('ContainerSchedule', () => {
  test('should render ContainerSchedule title component', () => {
    const wrapper = renderer.create(
      <Schedule.DayContainer
        data={data}
        daysPosition={'monday'}
        daysVal={'7:00 - 8:00'}
      />
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('should render ContainerSchedule title component', () => {
    const wrapper = renderer.create(
      <Schedule.DayContainer
        borderRightSchedule
        data={data}
        daysPosition={'monday'}
        daysVal={'7:00 - 8:00'}
      />
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
