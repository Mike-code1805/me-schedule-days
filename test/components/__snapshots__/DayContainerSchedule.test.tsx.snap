// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContainerSchedule should render ContainerSchedule title component 1`] = `
<div
  className="container_schedule_lineTime_caps_mini_course"
  style={
    Object {
      "borderRight": "",
    }
  }
>
  <div
    className="containerSchedule"
    onClick={[Function]}
    style={
      Object {
        "background": "red",
      }
    }
  >
    <h1
      className="containerSchedule_text"
    >
      Procesamiento de Lenguaje Natural
    </h1>
  </div>
</div>
`;

exports[`ContainerSchedule should render ContainerSchedule title component 2`] = `
<div
  className="container_schedule_lineTime_caps_mini_course"
  style={
    Object {
      "borderRight": "1px solid #D4D4D4",
    }
  }
>
  <div
    className="containerSchedule"
    onClick={[Function]}
    style={
      Object {
        "background": "red",
      }
    }
  >
    <h1
      className="containerSchedule_text"
    >
      Procesamiento de Lenguaje Natural
    </h1>
  </div>
</div>
`;
